### PR TITLE
Fix mbedtls entropy

### PR DESCRIPTION
**Pull request by isage**
Use patch instead of sed and implement entropy via getentropy on vita

### CODE REVIEW

**Review by sharkwouter — APPROVED** [review]
This looks like a nice improvement. I would suggest trying to upstream this change, as it is a very minimal one. In that case you should probably address my little nitpick, otherwise you can ignore it and go ahead and merge.

File: mbedtls/vita.patch [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+     *olen = 0;
+diff --git a/library/net_sockets.c b/library/net_sockets.c
+index d1700f3..af7379e 100644
+--- a/library/net_sockets.c
++++ b/library/net_sockets.c
+@@ -33,7 +33,7 @@
+ 
+ #if !defined(unix) && !defined(__unix__) && !defined(__unix) && \
+     !defined(__APPLE__) && !defined(_WIN32) && !defined(__QNXNTO__) && \
+-    !defined(__HAIKU__) && !defined(__midipix__)
```
[the remark this answers is not in the document]
Comment (isage) [reply mbedtls/vita.patch:67]: Yeah, that's the result of original sed invocation, meh. Fixed.

**Comment by isage** [comment]
Feel free to upstream, because i don't intend to.

**Comment by sharkwouter** [comment]
Okay, I made a PR upstream: https://github.com/Mbed-TLS/mbedtls/pull/6443

**Comment by scribam** [comment]
Is this PR ready? Can it be merged?

**Comment by isage** [comment]
It was ready a long time ago. And no, now it needs rebasing and updating patch (or upstreaming)

**Comment by d3m3vilurr** [comment]
closed by #273

**Pull request closed without merging**